webpack.config.js

```diff
@@ -21,46 +20,204 @@ module.exports = {
-            { test: /\.js$/, use: "babel-loader", include: path.resolve(__dirname, 'src') },
             {
-                test: /\.wasm$/,
-                loader: "file-loader",
-                type: "javascript/auto", // https://github.com/webpack/webpack/issues/6725
+                test: /\.(ts|js)x?$/,
+                exclude: /node_modules/,
+                loader: 'babel-loader',
                 options: {
-                    name: '[name].[hash:7].[ext]',
-                    outputPath: '.',
-                },
+                    // We have the babel config here rather than a file to help
+                    // make it clearer what we're compiling to.
+                    cacheDirectory: true,
+                    sourceMaps: true,
+                    presets: [
+                        ["@babel/preset-env", {
+                            "targets": {
+                                "browsers": [
+                                    "last 2 versions",
+                                ]
+                            },
+                        }],
+                        "@babel/preset-typescript",
+                        "@babel/preset-flow",
+                        "@babel/preset-react",
+                    ],
+                    plugins: [
+                        // Most of these plugins are for the react-sdk to make itself
+                        // work correctly.
+                        ["@babel/plugin-proposal-decorators", {"legacy": true}],
+                        "@babel/plugin-proposal-export-default-from",
+                        "@babel/plugin-proposal-numeric-separator",
+                        "@babel/plugin-proposal-class-properties",
+                        "@babel/plugin-proposal-object-rest-spread",
+                        "@babel/plugin-transform-flow-comments",
+                        "@babel/plugin-syntax-dynamic-import",
+                        "@babel/plugin-transform-runtime"
+                    ],
+                }
+            },
+            {
+                test: /\.css$/,
+                use: [
+                    MiniCssExtractPlugin.loader,
+                    {
+                        loader: 'css-loader',
+                        options: {
+                            importLoaders: 1,
+                            sourceMap: true,
+                        }
+                    },
+                    {
+                        loader: 'postcss-loader',
+                        ident: 'postcss',
+                        options: {
+                            sourceMap: true,
+                            plugins: () => [
+                                // Note that we use significantly fewer plugins on the plain
+                                // CSS parser. If we start to parse plain CSS, we end with all
+                                // kinds of nasty problems (like stylesheets not loading).
```

Comment:
wait, why are we now running plain css through postcss at all?

Reply:
So this is fun... `postcss` for plain css is so the scss sheets we have can reference plain css without exploding, which is something we do. It also acts as the babel for css, bringing things down to a safe place that we can work from. When it comes to bundling, both scss and css are mashed together by the chunking options way up at the top, allowing them to be exported.

Oh, and to make things worse: if we don't explicitly run it through postcss then it assumes that the css coming in is scss and it *really* doesn't like that (it looked like it sanity checked it before parsing it, despite being compatible).

Reply:
Huh, ok - this one is probably worth mentioning in the comment.